pkg/util/unixsock.go

```diff
@@ -0,0 +1,66 @@
+package util
+
+import (
+	"io"
+	"log"
+	"net"
+	"strings"
+)
+
+// IstgtUctlUnxpath is the storage path for the UNIX domain socket from istgt
+const (
+	IstgtUctlUnxpath = "/var/run/istgt_ctl_sock"
+	EndOfStream      = "\r\n"
+	Prefix           = "iSCSI Target Controller version"
+)
+
+//Reader reads the response from unix domain socket
+func Reader(r io.Reader) []byte {
+	resp := make([]byte, 4*1024)
+	buf := make([]byte, 1024)
+	for {
+		n, err := r.Read(buf[:])
+		if err != nil {
+			println("return resp : ", string(resp))
+			return resp
+		}
+		println("Client got:", string(buf[0:n]))
+		if strings.HasPrefix(string(buf[0:n]), Prefix) {
+			continue
```

Comment:
Is it possible that the entire response is read in one read iteration? If yes, the resp should be populated with the entire buffer contents.